### PR TITLE
ipex intel extension for pytorch integration

### PR DESCRIPTION
**Pull request by pacman100**
### What does this PR do?
1. Implements feature request #700

### CODE REVIEW

**Comment by liangan1** [comment]
Thanks @pacman100. 
@tangleiintel pls refine the [RFC#700](https://github.com/huggingface/accelerate/issues/700) and validate this PR.

File: src/accelerate/utils/imports.py [hunk 1]
```diff
@@ -125,3 +128,26 @@ def is_sagemaker_available():
...
+
+    _torch_version = importlib_metadata.version("torch")
+    if importlib.util.find_spec("intel_extension_for_pytorch") is None:
+        return False
+    _ipex_version = "N/A"
+    try:
+        _ipex_version = importlib_metadata.version("intel_extension_for_pytorch")
+    except importlib_metadata.PackageNotFoundError:
+        return False
+    torch_major_and_minor = get_major_and_minor_from_version()
```
Comment (tangleintel) [inline src/accelerate/utils/imports.py:145]: Should '_torch_version' be passed to the inner function get_major_and_minor_from_version()?

**Comment by Ankur-singh** [comment]
any updates on this??

**Comment by pacman100** [comment]
No action for now as @tangleintel was supposed to pick this up, make remaining changes, test it and then reraise the PR. If many people want this as feature, it will be accordingly prioritised.

**Comment by jianan-gu** [comment]
Hi @pacman100  @Ankur-singh, 
Thanks for asking this,
We are working in progress intenally for this feature, and expect to reraise this PR within this month.

**Comment by jianan-gu** [comment]
https://github.com/pacman100/accelerate/pull/1 to refine JIT enabling

**Comment by jianan-gu** [comment]
@pacman100 Hi, this PR is ready for review now~ let's go ahead to make it. Thanks.

**Comment by sywangyi** [comment]
Hi, @sgugger, could you help review it?

**Comment by sgugger** [comment]
Could you fix the conflicts with main first please?

**Comment by jianan-gu** [comment]
Hi @pacman100 , 
Is it possible that you could help highlight the conflicts with the main branch?
I would like to fix it but could not see the detail (since i do have not the write access)
Many thanks!

**Comment by sgugger** [comment]
@jianan-gu You do not need write access to make a rebase/merge with the main branch.

**Comment by sywangyi** [comment]
@sgugger I rebase it and create a new PR
https://github.com/huggingface/accelerate/pull/1255

**Pull request closed without merging**